### PR TITLE
Home feature

### PR DESCRIPTION
**Pull request by ahmed-sala**
add: home feature backend
add: home screen design

### CODE REVIEW

File: lib/src/presentation/managers/home/home_state.dart [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+
+sealed class HomeState {
+  const HomeState();
+}
+
+class HomeStateLoading extends HomeState {
+  const HomeStateLoading();
+}
+
+class HomeStateSuccess extends HomeState {
```
Comment (ali72-20) [inline lib/src/presentation/managers/home/home_state.dart:14]: we can make object called products list contains three lists according to make there list. 
```
class ProductsLists{
 final List<ProductsModel>? products;
  final List<BestSellerModel>? bestSeller;
  final List<OccasionsModel>? occasions;
 // add constructor
  
}

File: lib/src/presentation/widgets/home/app_bar/location_widget.dart [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+  @override
+  Widget build(BuildContext context) {
+    return const Row(
+      crossAxisAlignment: CrossAxisAlignment.center,
+      children: [
+        ImageIcon(
+          AssetImage(AppIcons.locationIcon),
+          size: 20,
+        ),
+        SizedBox(width: 8),
```
Comment (ali72-20) [inline lib/src/presentation/widgets/home/app_bar/location_widget.dart:18]: try use vertical or horizontal space

File: lib/src/presentation/widgets/home/app_bar/search_textfeild.dart [hunk 3]
```diff
@@ -0,0 +1,28 @@
+import 'package:flower_app/core/styles/colors/app_colors.dart';
+import 'package:flutter/material.dart';
+import 'package:flutter_gen/gen_l10n/app_localizations.dart';
+
+class SearchTextfeild extends StatelessWidget {
+  const SearchTextfeild({super.key});
+
+  @override
+  Widget build(BuildContext context) {
+    return TextField(
```
Comment (ali72-20) [inline lib/src/presentation/widgets/home/app_bar/search_textfeild.dart:10]: There is a custom Text input field try use it

File: lib/src/presentation/widgets/home/category/category_card.dart [hunk 4]
```diff
@@ -0,0 +1,28 @@
...
+
+class CategoryCard extends StatelessWidget {
+  const CategoryCard({super.key, required this.imageAsset});
+
+  final String imageAsset;
+
+  @override
+  Widget build(BuildContext context) {
+    return Container(
+      width: 68,
```
Comment (ali72-20) [inline lib/src/presentation/widgets/home/category/category_card.dart:13]: add h after hight value and w after width value for responsive ui

File: lib/src/presentation/widgets/home/products/discover_product_item.dart [hunk 5]
```diff
@@ -0,0 +1,87 @@
...
+                            borderRadius: BorderRadius.circular(8),
+                            side: const BorderSide(
+                              color: Colors.white,
+                              width: 1,
+                            ),
+                          ),
+                        ),
+                        child: Text(
+                          AppLocalizations.of(context)!.giftNow,
+                          style: TextStyle(
```
Comment (ali72-20) [inline lib/src/presentation/widgets/home/products/discover_product_item.dart:71]: try use app text styles class

File: lib/src/presentation/widgets/home/home_product_item.dart [hunk 6]
```diff
@@ -0,0 +1,45 @@
...
+            fontSize: price == null ? 16 : 14,
+            color: Colors.black87,
+            fontWeight: price == null ? FontWeight.bold : FontWeight.normal,
+          ),
+          textAlign: TextAlign.center,
+        ),
+        if (price != null) const SizedBox(height: 4),
+        if (price != null)
+          Text(
+            '$price EGP',
```
Comment (ali72-20) [inline lib/src/presentation/widgets/home/home_product_item.dart:34]: use local

File: lib/src/presentation/widgets/home/home_product_item.dart [hunk 7]
```diff
@@ -0,0 +1,45 @@
...
+  final num? price;
+
+  @override
+  Widget build(BuildContext context) {
+    return Column(
+      crossAxisAlignment: CrossAxisAlignment.start,
+      children: [
+        Image.network(
+          imageUrl,
+          width: 130,
```
Comment (ali72-20) [inline lib/src/presentation/widgets/home/home_product_item.dart:17]: w after wigth 
h after hight 
for responsive ui

**Review by ali72-20 — COMMENTED** [review]
solve this little comments

File: lib/src/presentation/widgets/home/app_bar/search_textfeild.dart (hunk 3 above)
Comment (ahmed-sala) [reply lib/src/presentation/widgets/home/app_bar/search_textfeild.dart:10]: in search it is different in design

File: lib/src/presentation/widgets/home/home_product_item.dart (hunk 6 above)
Comment (ahmed-sala) [reply lib/src/presentation/widgets/home/home_product_item.dart:34]: what is meaning of local

**Review by basharwalid — APPROVED** [review]

**Review by ali72-20 — APPROVED** [review]
done

**Pull request merged**